test/integration/stats_integration_test.cc

```diff
@@ -128,5 +136,59 @@ TEST_P(StatsIntegrationTest, WithTagSpecifierWithFixedValue) {
   EXPECT_EQ(live->tags()[0].value_, "xxx");
 }
 
+INSTANTIATE_TEST_SUITE_P(IpVersions, ClusterMemoryUtilization,
+                         testing::ValuesIn(TestEnvironment::getIpVersionsForTest()),
+                         TestUtility::ipTestParamsToString);
+
+//uint64_t memoryConsumedWithClusters(int num_clusters, bool allow_stats) {
+//    config_helper_.addConfigModifier([&](envoy::config::bootstrap::v2::Bootstrap& bootstrap) {
+//        if (!allow_stats) {
+//            bootstrap.mutable_stats_config()->mutable_stats_matcher()->set_reject_all(true);
+//        }
+//        for (int i = 1; i < num_clusters; i++) {
+//            auto* c = bootstrap.mutable_static_resources()->add_clusters();
+//            c->set_name(fmt::format("cluster_{}", i));
+//        }
+//    });
+//
+//};
+
+// TODO(cmluciano) Refactor once envoyproxy/envoy#5624 is solved
+// TODO(cmluciano) Add options to measure multiple workers
+TEST_P(ClusterMemoryUtilization, MemoryLargeClusterSizeWithStats) {
+    const size_t start_mem = Memory::Stats::totalCurrentlyAllocated() / 1000;
+    if (start_mem == 0) {
+        // Skip this test for platforms where we can't measure memory.
+        return;
+    }
+
+    auto IpVersions = TestBaseWithParam<Network::Address::IpVersion>::GetParam();
+
+    // HTTP_PROXY_CONFIG with 1 cluster
+    auto* t1 = new BaseIntegrationTest(IpVersions);
+    t1->initialize();
+    size_t m1 = Memory::Stats::totalCurrentlyAllocated();
+    EXPECT_LT(start_mem, m1);
+    EXPECT_EQ(m1, 57000);
+    delete t1;
```

Comment:
t1.reset(nullptr), or better yet, put this block in a scope or better still, a helper method it can share with the block below.